ebu_tt_live/carriage/filesystem.py

```diff
@@ -115,44 +132,74 @@ def resume_producing(self):
         # for times in the manifest file depending on your time base.
         if sequence_number is None:
             # This means that it isn't a document. It can be a message.
             self._msg_counter += 1
-            filename = '{}_msg_{}.xml'.format(sequence_identifier, self._msg_counter)
+            filename = self._message_file_name_pattern.format(
+                counter=self._msg_counter, 
+                sequence_identifier=sequence_identifier)
+            #filename = '{}_msg_{}.xml'.format(sequence_identifier, self._msg_counter)
         else:
-            filename = '{}_{}.xml'.format(sequence_identifier, sequence_number)
+            #how do i tell the difference between a message and an EBU-TT-D document?
+            #maybe the answer is to use the message format for EBU-TT-D documents!
+            filename = self._file_name_pattern.format(
+                counter=sequence_number, 
+                sequence_identifier=sequence_identifier)
+            #filename = '{}_{}.xml'.format(sequence_identifier, sequence_number)
+            
+        # consider using different classes or functions to do the document writing,
+        # depending on the settings of suppress_manifest and rotating_buf etc that
+        # can be selected once at the beginning and dereferenced rather than repeating
+        # if statements.
         filepath = os.path.join(self._dirpath, filename)
-        with open(filepath, 'w') as f:
-            f.write(data)
-        time_base = time_base
-        availability_time = availability_time
-        if delay is not None:
-            availability_time += timedelta(seconds=delay)
-        new_manifest_line = '{},{}\n'.format(timedelta_to_str_manifest(availability_time), filename)
-        self._manifest_content += new_manifest_line
-        with open(self._manifest_path, 'a') as f:
-            f.write(new_manifest_line)
+        with open(filepath, 'w') as destfile:
+            destfile.write(data)
+            destfile.flush()
+                        
+        # If we're running a rotating buffer remember this file for possible deletion later
+        if self._circular_buf_size > 0 :
+            self._circular_buf.append(filepath)
+            
+        if not self._suppress_manifest :
+            # Work out what time to put in the manifest file
+            
+            # NOTE: This is nasty
+            availability_time = self.check_availability_time(
+                sequence_identifier=sequence_identifier,
+                time_base=time_base,
+                clock_mode=clock_mode,
+                availability_time=availability_time
+            )
+
+            if availability_time is None:
+                # This is a possibility with a live messages as first document. They don't contain enough timing info.
+                log.warning(
+                    FS_MISSING_AVAILABILITY.format(sequence_identifier=sequence_identifier,
```

Comment:
Stick a new line in between `.format(` and `sequence_identifier=sequence_identifier,`